### PR TITLE
Improve documentation for client/metrics pkg

### PR DESCRIPTION
**Pull request by tmrts**
Adds documenation for the `client/metrics` package and it's exposed members.

### CODE REVIEW

File: pkg/client/metrics/metrics.go [hunk 1]
```diff
@@ -26,6 +27,8 @@ import (
 const restClientSubsystem = "rest_client"
 
 var (
+	// RequestLatency is a SummaryVec to be used while registering
+	// metrics to Prometheus.
```
Comment (fgrzadkowski) [inline pkg/client/metrics/metrics.go]: I don't understand this comment. What do you mean?

File: pkg/client/metrics/metrics.go [hunk 2]
```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// Package metrics provides utilities for registering metrics to Prometheus.
```
Comment (fgrzadkowski) [inline pkg/client/metrics/metrics.go]: ... for registering client metrics to Prometheus.

**Comment by tmrts** [comment]
@fgrzadkowski clarified SummaryVec, if it looks good I'll squash it.

**Comment by fgrzadkowski** [comment]
@tmrts LGTM Please squash and I'll add proper labels.

**Comment by tmrts** [comment]
@fgrzadkowski I've squashed the commit. Pinging just in case you didn't see the notice yesterday

**Comment by fgrzadkowski** [comment]
@tmrts Yeah, I missed yesterday. Sorry for the delay and thanks for the fix!

**Pull request merged**